userspace/libscap/scap_bpf.c

```diff
@@ -140,6 +140,19 @@ static int bpf_map_create(enum bpf_map_type map_type,
 	return sys_bpf(BPF_MAP_CREATE, &attr, sizeof(attr));
 }
 
+static int bpf_map_freeze(int fd)
+{
+	union bpf_attr attr;
+
+	bzero(&attr, sizeof(attr));
+
+	attr.map_fd = fd;
+
+	/* Do not check for errors as BPF_MAP_FREEZE was introduced in kernel 5.2 */
```

Comment:
I think we'll need to do this only for >= 5.2 kernels and have it be a no-op for other kernels. (I'm thinking of redhat/centos where the kernels are very old).

Reply:
@mstemm indeed, the syscall on older kernels is not implemented thus this is already a no-op (the syscall will be rejected); that's why i do not check its return code.
If you don't like this, i can `#ifdef` the code.

Reply:
I think it would be better to #ifdef then rely on it being a no-op.

Reply:
Done!